src/services/document_service.py

```diff
@@ -242,7 +242,7 @@ def get_document(self, address: Address, depth: int = 0) -> Node | ListNode:
 
     def remove(self, address: Address) -> None:
         node = self.get_document(address)
-        if node.parent and not node.attribute.is_optional:
+        if node.parent and not node.attribute.is_optional and not node.parent.is_array():
```

Comment:
Question: 
What if there is a required element in array? Would that not pass through this test? 

Do we not want to check if the element itself in the array is optional, not whether the parent array is optional?

Reply:
Do the children of array inherit the 'optional' status of parent array?

Reply:
We have no concept to say that items in a list is required or not.

Children of arrays to inherit the "node.attribute" instance. Added a "is_optional" to NodeBase to make this check a bit more intuitive.